tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
```

Comment:
Why are we adding a root `tsconfig`? All of the packages extend the `tsconfig.settings` file that has all of these values? I am not opposed to adding all of these extra files, but I would like to understand what improvement this provides. It has been an ongoing challenge getting lerna/yarn/typescript/eslint/vscode/webpack all working together.

Reply:
I had to add all of these to get projectOptions working properly with `no-floating-promises`. I tried to find a way to get everything working without them, but no dice 😢